src/sip/transp.c

```diff
@@ -145,16 +151,33 @@ static const struct sip_transport *transp_find(struct sip *sip,
 					       int af, const struct sa *dst)
 {
 	struct le *le;
-	(void)dst;
+	struct sa dsttmp;
 
 	for (le = sip->transpl.head; le; le = le->next) {
 
 		const struct sip_transport *transp = le->data;
+		const struct sa *laddr = &transp->laddr;
+		struct sa src;
 
 		if (transp->tp != tp)
 			continue;
 
-		if (af != AF_UNSPEC && sa_af(&transp->laddr) != af)
+		if (af != AF_UNSPEC && sa_af(laddr) != af)
+			continue;
+
+		if (!sa_isset(dst, SA_ADDR))
+			return transp;
+
+		if (sa_is_linklocal(laddr) != sa_is_linklocal(dst))
+			continue;
+
+		sa_cpy(&dsttmp, dst);
+		sa_set_scopeid(&dsttmp, sa_scopeid(laddr));
+
+		if (net_dst_source_addr_get(&dsttmp, &src))
+			continue;
```

Comment:
is it so that the route check is only done for link-local ?

if so the baresip PR cannot really be applied, as it assumes that multiple addr
for all address types work.

Reply:
no, the route check is done for each `laddr`.